unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/GltfContainer/GltfContainerCollidersHandler.cs

```diff
@@ -0,0 +1,180 @@
+using DCL.Configuration;
+using DCL.Helpers;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace DCL.ECSComponents
+{
+    public class GltfContainerCollidersHandler
+    {
+        private readonly List<Collider> invisibleMeshesColliders = new List<Collider>(10); // colliders from `_collider`
+        private readonly List<Collider> visibleMeshesColliders = new List<Collider>(10); // colliders from renderers
+        private readonly List<Collider> visibleMeshesInvalidColliders = new List<Collider>(10); // colliders created from sdk6 but not valid for sdk7
+        private IReadOnlyCollection<Renderer> cachedRenderers;
+
+        // We restore all modifications on colliders to avoid issues with sdk6 GLTFShape
+        public void CleanUp()
+        {
+            for (int i = 0; i < invisibleMeshesColliders.Count; i++)
+            {
+                Collider collider = invisibleMeshesColliders[i];
+                collider.enabled = true;
+                collider.gameObject.layer = PhysicsLayers.characterOnlyLayer;
+            }
+
+            for (int i = 0; i < visibleMeshesColliders.Count; i++)
+            {
+                Collider collider = visibleMeshesColliders[i];
+                collider.enabled = true;
+                collider.gameObject.layer = PhysicsLayers.onPointerEventLayer;
+            }
+
+            for (int i = 0; i < visibleMeshesInvalidColliders.Count; i++)
+            {
+                Collider collider = visibleMeshesInvalidColliders[i];
+                collider.enabled = true;
+                collider.gameObject.layer = PhysicsLayers.onPointerEventLayer;
+            }
+
+            invisibleMeshesColliders.Clear();
+            visibleMeshesColliders.Clear();
+            visibleMeshesInvalidColliders.Clear();
+        }
+
+        // Since gltf are modified when creating colliders and stored in cache library with those modifications
+        // we check if colliders already exists and disable them
+        // otherwise we create them (disabled)
+        public IReadOnlyList<Collider> InitInvisibleMeshesColliders(IReadOnlyList<MeshFilter> meshFilters)
+        {
+            invisibleMeshesColliders.Clear();
+
+            for (int i = 0; i < meshFilters.Count; i++)
+            {
+                if (meshFilters[i].gameObject.layer == PhysicsLayers.characterOnlyLayer)
+                {
+                    Collider collider = meshFilters[i].gameObject.GetComponent<Collider>();
```

Comment:
Shouldn't the `AddComponent<MeshCollider>()` happen here if `collider == null` ? (`TryGetComponent<MeshCollider>()` could also be used to know if the component was found)

Reply:
I was making the assumption that if gameobject's layer is `PhysicsLayers.characterOnlyLayer` then it must be a collider,
but I forgot that sdk6 can also place a collider on the default layer, so my assumptions it's not entirely correct, so I must change that

Reply:
changed